partiql-parser/src/lalr/lexer.rs

```diff
@@ -655,41 +695,43 @@ mod tests {
 
     #[test]
     fn ion_simple() {
-        let ion_value = r#" `{'a':1,  'b':1}` "#;
+        let ion_value = r#" `{'input':1,  'b':1}` "#;
+
+        let mut offset_tracker = LineOffsetTracker::default();
+        let ion_lexer = EmbeddedIonLexer::new(ion_value.trim(), &mut offset_tracker);
```

Comment:
nit: Does it make sense to trim the value inside the `new`?

Reply:
It's a bit odd-looking, but meant to be for brevity in this test.

Rust's [str.trim](https://doc.rust-lang.org/std/primitive.str.html#method.trim) isn't a destructive method, rather it just just slices the `&str` to remove the leading and trailing whitespace. In this instance, I could have replaced it with `&ion_value[1..ion_value.len() - 1]`.

As to why we need to remove the whitespace, the `EmbeddedIonLexer` assumes that it is called at the backtick that begins the embedding.

Reply:
> As to why we need to remove the whitespace, the EmbeddedIonLexer assumes that it is called at the backtick that begins the embedding.

Gotcha, so I actually meant could `EmbeddedIonLexer` do the trim rather than making this assumption.